sklearn/neighbors/_nca.py

```diff
@@ -334,8 +349,6 @@ def _validate_params(self, X, y):
 
         # Check the preferred dimensionality of the projected space
         if self.n_components is not None:
-            check_scalar(self.n_components, "n_components", numbers.Integral, min_val=1)
-
             if self.n_components > X.shape[1]:
```

Comment:
These two `if` statements can be combined now.

Reply:
good catch!